### ISSUE TITLE
 Need Help: Compilation problem with Shogun

### ISSUE THREAD

**Issue by UsmanSyed** [shogun-toolbox/shogun#1995]
Hi,
 I am trying to compile Shogun and while configuring it I enables python and
octave variants as well. Now it compiles uptil 84% and then an error appears
while compilation uses octave files. Specifically it is unable to find
"config.h". Here is the specific error:

[ 84%] Building CXX object
src/interfaces/octave_modular/CMakeFiles/octave_modular.dir/modshogunOCTAVE_w
rap.cxx.o
In file included from /home/mani/Installations/shogun-
3.2.0/build/src/interfaces/octave_modular/modshogunOCTAVE_wrap.cxx:153:0:
/opt/octave3.8/include/octave-3.8.0/octave/oct.h:31:20: fatal error:
config.h: No such file or directory
compilation terminated.
make[2]: ***
[src/interfaces/octave_modular/CMakeFiles/octave_modular.dir/modshogunOCTAVE_
wrap.cxx.o] Error 1
make[1]: ***
[src/interfaces/octave_modular/CMakeFiles/octave_modular.dir/all] Error 2
make: *** [all] Error 2

I compiled octave myself (just a week ago) and its working fine. Earlier I was having a version of octave from my ubuntu distribution. I am using Ubuntu 12.10 and it contains octave3.6 while shogun needed octave3.8. So I compiled it my self. 

Please help me resolve this issue.

**Comment by beew** [comment]
It is a bit late but it appears that this still has not been resolved. I tried to build shogun 4.0 octave modular against octave 3.8.2 on Ubuntu 14.04 and got the same error
".. fatal error: config.h: No such file or directory"

Turned out it was because of this bug http://savannah.gnu.org/bugs/?41027, or rather, the attempt to fix it. Shogun octave built successfully after rebuilding octave with the patch from the thread above.
